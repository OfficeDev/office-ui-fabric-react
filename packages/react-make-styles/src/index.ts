export { ax, createDOMRenderer } from '@fluentui/make-styles';

export { makeStylesCompat } from './makeStyles';
export { makeStaticStyles } from './makeStaticStyles';
