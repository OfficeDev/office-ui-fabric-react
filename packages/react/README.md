# @uifabric/react

**React components for [Office UI Fabric React](https://dev.microsoft.com/fabric)**

These are not production-ready components and **should never be used in product**. This space is useful for testing new components whose APIs might change before final release.

To import React components:

```js
import { ComponentName } from '@uifabric/react';
```

Once the React component graduates to a production release, the component will be available at:

```js
import { ComponentName } from 'office-ui-fabric-react';
```
