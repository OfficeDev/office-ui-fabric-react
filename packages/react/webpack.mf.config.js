const path = require('path');
const fs = require('fs');
const webpack = require('webpack');
const getResolveAlias = require('@fluentui/scripts/webpack/getResolveAlias');
const { createServeConfig } = require('@fluentui/scripts/webpack/webpack-resources');
const { webpackMerge } = require('just-scripts');

const shared = {
  react: { singleton: true },
  'react-dom': { singleton: true },
};

const rootComponents = fs
  .readdirSync(path.join(__dirname, 'src'))
  .filter(filename => filename[0].toUpperCase() === filename[0])
  .map(filename => filename.replace(/\.tsx?/, ''));
const rootComponentsExposes = {};
for (const component of rootComponents) {
  rootComponentsExposes[`./lib/${component}`] = `@fluentui/react/src/${component}`;
}

// TODO: fork-ts-checker-webpack-plugin needs to be taught about the resolve alias -> pathmapping in order for the type checking to work
process.env.SKIP_TYPECHECK = true;

const myConfig = {
  output: {},
  entry: './empty-entrypoint.js',
  mode: 'development',
  devtool: 'cheap-module-source-map',
  optimization: {
    concatenateModules: false,
  },
  module: {
    rules: [],
  },
  plugins: [
    new webpack.container.ModuleFederationPlugin({
      name: 'fluentuiReact',
      filename: 'remoteEntry.js',
      exposes: {
        ...rootComponentsExposes,
        './lib/compat/Button': '@fluentui/react/src/compat/Button',
        '.': '@fluentui/react/src/index',
        './lib/index': '@fluentui/react/src/index',
      },
      shared,
    }),
  ],
  resolve: {
    alias: getResolveAlias(),
  },
  devServer: {
    port: 2345,
    static: './dist',
  },
};

const metaConfig = webpackMerge.merge(createServeConfig(), myConfig);

module.exports = metaConfig;
