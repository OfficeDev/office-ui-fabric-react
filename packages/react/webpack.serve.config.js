const path = require('path');
const resources = require('@uifabric/build/webpack/webpack-resources');

module.exports = resources.createServeConfig({
  entry: './src/demo/index.tsx',

  output: {
    filename: 'demo-app.js',
  },

  externals: {
    react: 'React',
    'react-dom': 'ReactDOM',
  },

  resolve: {
    alias: {
      '@fluentui/react/src': path.join(__dirname, 'src'),
      '@fluentui/react/lib': path.join(__dirname, 'lib'),
      '@fluentui/react': path.join(__dirname, 'lib'),
      'Props.ts.js': 'Props',
      'Example.tsx.js': 'Example',
    },
  },
});
