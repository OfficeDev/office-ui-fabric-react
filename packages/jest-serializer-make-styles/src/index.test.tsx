import '@testing-library/jest-dom';
import * as React from 'react';
import { print, test } from './index';
import { render } from '@testing-library/react'; // (or /dom, /vue, ...)
import { makeStyles, ax } from '@fluentui/react-make-styles';
import * as renderer from 'react-test-renderer';

const useStyles1 = makeStyles({
  paddingLeft: {
    paddingLeft: '10px',
  },
});
const useStyles2 = makeStyles({
  paddingRight: {
    paddingRight: '20px',
  },
});
const useStyles3 = makeStyles({
  display: {
    display: 'none',
  },
});

const Test = () => {
  const styles1 = useStyles1();
  const styles2 = useStyles2();
  const styles3 = useStyles3();
  return (
    <div
      data-testid="test"
      className={ax('static-class', styles1.paddingLeft, styles2.paddingRight, styles3.display)}
    />
  );
};

describe('jest-serializer-make-styles', () => {
  it('should check styles', () => {
    expect(render(<Test />).getByTestId('test')).toHaveStyle({
      display: 'none',
      paddingLeft: '10px',
      paddingRight: '20px',
    });
  });

  it('renders a default state', () => {
    const component = renderer.create(<Test />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

expect.addSnapshotSerializer({ print, test });
