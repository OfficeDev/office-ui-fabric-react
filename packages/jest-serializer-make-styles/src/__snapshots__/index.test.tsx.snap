// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-serializer-make-styles renders a default state 1`] = `
<div
  className="static-class"
  data-testid="test"
/>
`;
